keys/keybase.go

```diff
@@ -7,62 +7,111 @@ import (
 	"strings"
 
 	"github.com/pkg/errors"
-	crypto "github.com/tendermint/go-crypto"
-	"github.com/tendermint/go-crypto/keys/words"
+	"github.com/tendermint/go-crypto"
+	"github.com/tendermint/go-crypto/keys/bip39"
+	"github.com/tendermint/go-crypto/keys/hd"
 	dbm "github.com/tendermint/tmlibs/db"
 )
 
-// dbKeybase combines encyption and storage implementation to provide
+var _ Keybase = dbKeybase{}
+
+// Language is a language to create the BIP 39 mnemonic in.
+// Currently, only english is supported though.
+// Find a list of all supported languages in the BIP 39 spec (word lists).
+type Language int
+
+const (
+	// English is the default language to create a mnemonic.
+	// It is the only supported language by this package.
+	English Language = iota
```

Comment:
Would be nice to have the first value at `iota + 1` to prevent accidental zero value passing.

Reply:
Should we consider prefixing with `Language`.

Reply:
iota+1 it is 👍 

> Should we consider prefixing with Language.

You mean s/English/LanguageEnglish/ ? Why? Isn't that a bit redundant?

Reply:
There is a case to make for either on the caller side it's a bit more clear when reading, yet the type system should inform that already. Could be either way, in general it's good to prefix if the namespace of the package gets a bit more crowded.